src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -647,14 +680,32 @@ private bool IsNonStringArray(JsonPatchProperty patchProperty)
             ConversionResult result,
             T objectToApplyTo,
             Operation<T> operation,
-            string path)
+            string path,
+            Action<string> action)
         {
             if (!result.CanBeConverted)
             {
-                throw new JsonPatchException<T>(
+                LogError(
+                    objectToApplyTo,
                     operation,
                     Resources.FormatInvalidValueForProperty(result.ConvertedInstance, path),
-                    objectToApplyTo);
+                    action);
+            }
+        }
+
+        private void LogError(
+            T objectToApplyTo,
+            Operation<T> operation,
+            string errorMessage,
+            Action<string> action = null)
+        {
+            if (action != null)
+            {
+                action(errorMessage);
+            }
+            else
+            {
+                throw new JsonPatchException<T>(operation, errorMessage, objectToApplyTo);
```

Comment:
Should we introduce a type for this (`JsonPatchError`) rather than just a string?